src/platform_impl/web/event_loop/runner.rs

```diff
@@ -227,7 +226,9 @@ impl<T: 'static> Shared<T> {
         // If the runner doesn't exist and this method recurses, it will recurse infinitely
         if !is_closed && self.0.runner.borrow().is_some() {
             // Take an event out of the queue and handle it
-            if let Some(event) = self.0.events.borrow_mut().pop_front() {
+            // Make sure not to let the borrow_mut live during the next handle_event
+            let event = { self.0.events.borrow_mut().pop_front() };
+            if let Some(event) = event {
```

Comment:
I am new to Rust, but this fix looks like it shouldn't be necessary.
Is the `borrow_mut` alive in the whole scope of the `if let`?

Reply:
That's what I assumed also, which is why I ended up writing the bug in the first place 😛 

I'd guess it has to do with the scope the value is alive for? Even if a value isn't used, I believe the drop is performed at the end of the scope it lives in. Therefore the old borrow lasted until the end of the `if let` (or maybe even the outer block) even though it could have been dropped earlier.

Reply:
https://users.rust-lang.org/t/curious-scope-rules-when-using-if-let/1858/2